lib/rubocop/cop/lint/out_of_range_regexp_ref.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cops looks for out of range referencing for Regexp, as while capturing groups out of
+      # out of range reference always returns nil.
+
```

Comment:
I just noticed that the doc file is incomplete, because you have a missing `#` here